File: seed/php-sdk/trace/src/V2/V3/Problem/Types/GetBasicSolutionFileRequest.php
<?php

namespace Seed\V2\V3\Problem\Types;

use Seed\Core\Json\JsonSerializableType;
use Seed\Core\Json\JsonProperty;

class GetBasicSolutionFileRequest extends JsonSerializableType
{
    /**
     * @var string $methodName
     */
    #[JsonProperty('methodName')]
    public string $methodName;

    /**
     * @var NonVoidFunctionSignature $signature
     */
    #[JsonProperty('signature')]
    public NonVoidFunctionSignature $signature;

    /**
     * @param array{
     *   methodName: string,
     *   signature: NonVoidFunctionSignature,
     * } $values
     */
    public function __construct(
        array $values,
    )
    {
        $this->methodName = $values['methodName'];$this->signature = $values['signature'];
    }

    /**
     * @return string
     */
    public function __toString(): string {
        return $this->toJson();
    }
}

File: seed/ruby-sdk-v2/trace/lib/seed/v_2/v_3/problem/get_basic_solution_file_request.rb
# frozen_string_literal: true

module Seed
    module Types
        class GetBasicSolutionFileRequest < Internal::Types::Model
            field :method_name, String, optional: false, nullable: false
            field :signature, Seed::V2::V3::Problem::NonVoidFunctionSignature, optional: false, nullable: false

    end
end
